account_credit_control/models/credit_control_policy.py

```diff
@@ -0,0 +1,416 @@
+        cr.execute(
+            "SELECT move_line_id FROM credit_control_line"
+            "    WHERE policy_id != %s and move_line_id in %s"
+            "    AND manually_overridden IS NOT True",
+            (self.id, tuple(lines.ids)),
+        )
+        res = cr.fetchall()
+        if res:
+            return different_lines.browse([row[0] for row in res])
+        return different_lines
+
+    def check_policy_against_account(self, account):
+        """Ensure that the policy corresponds to account relation"""
+        allowed = self.search(
+            ["|", ("account_ids", "in", account.ids), ("do_nothing", "=", True)]
+        )
+        if self not in allowed:
+            allowed_names = "\n".join(x.name for x in allowed)
+            raise UserError(
+                _(
+                    "You can only use a policy set on "
+                    "account %(account_name)s.\n"
+                    "Please choose one of the following "
+                    "policies:\n %(allowed_names)s"
+                )
+                % {"account_name": account.name, "allowed_names": allowed_names}
+            )
+        return True
+
+    def _generate_credit_lines(self, credit_control_run, default_lines_vals=None):
+        self.ensure_one()
+        controlling_date = credit_control_run.date
+        credit_line_model = self.env["credit.control.line"]
+        lines = self._get_move_lines_to_process(credit_control_run)
+        manual_lines = self._lines_different_policy(lines)
+        lines -= manual_lines
+        policy_lines_generated = credit_line_model
+        if lines:
+            # policy levels are sorted by level
+            # so iteration is in the correct order
+            create = policy_lines_generated.create_or_update_from_mv_lines
+            for level in reversed(self.level_ids):
+                level_lines = level.get_level_lines(controlling_date, lines)
+                policy_lines_generated += create(
+                    level_lines,
+                    level,
+                    controlling_date,
+                    credit_control_run.company_id,
+                    default_lines_vals=default_lines_vals,
+                )
+        if policy_lines_generated:
+            report = _(
+                'Policy "<b>%(name)s</b>" has generated <b>'
+                "%(len_policy_lines_generated)d Credit "
+                "Control Lines.</b><br/>"
+            ) % {
+                "name": self.name,
+                "len_policy_lines_generated": len(policy_lines_generated),
+            }
+        else:
+            report = (
+                _(
+                    'Policy "<b>%s</b>" has not generated any '
+                    "Credit Control Lines.<br/>"
+                )
+                % self.name
+            )
+        return (manual_lines, policy_lines_generated, report)
+
+    @api.model
+    def _name_search(self, name, domain=None, operator="ilike", limit=None, order=None):
```

Comment:
`_name_search` is gone, replaced by `_search_display_name`. A small test would be good as well, to prevent further regressions in the future.